kevm-pyk/src/kevm_pyk/utils.py

```diff
@@ -111,33 +114,50 @@ def kevm_apr_prove(
                 'EVM.return.success',
             ]
         )
-    prover: APRBMCProof | APRProver
+    prover: APRBMCProver | APRProver | EqualityProver
     if type(_apr_proof) is APRBMCProof:
         assert same_loop, f'BMC proof requires same_loop heuristic, but {same_loop} was supplied'
         prover = APRBMCProver(
             _apr_proof, is_terminal=is_terminal, extract_branches=extract_branches, same_loop=same_loop
         )
-    else:
+    elif type(_apr_proof) is APRProof:
         prover = APRProver(_apr_proof, is_terminal=is_terminal, extract_branches=extract_branches)
+    elif type(_apr_proof) is EqualityProof:
+        prover = EqualityProver(_apr_proof)
```

Comment:
We should probably have an `else:` case:

```suggestion
    elif type(_apr_proof) is EqualityProof:
        prover = EqualityProver(_apr_proof)
    else:
        raise ValueError(f'Do not know how to build prover for proof: {_apr_proo}')
```

And I guess we should rename variable `_apr_proof` to just `proof` or `_proof`.

Reply:
Changed this, and changed `kevm_apr_prove` to just `kevm_prove`.